src/scanner/Scanner.ts

```diff
@@ -157,13 +160,16 @@ export class Scanner {
     let localPath = scanningStrategy.localPath;
 
     if (!isOnline) {
+      this.repositoryConfig.localScanning = true;
```

Comment:
Don't mix `repositoryConfig` with `ScanningStrategy`. We could add more detailed info to scanningStrategy instead of repositoryConfig. You disrupt a separation of logic in our containers.